### PR TITLE
Prevent users from creating lessons with no notes #47

### PR DESCRIPTION
**Pull request by edXmO**
Hi! This is my first pull request so any advice is welcomed. I hope that i followed the guidelines and hope I didnt mess anything up. 

Fixed the problem of adding empty lesson notes, also I considered that lesson notes containing only whitespaces also satisfy the issue so I included that case as well. 

You can only add lesson notes that contain a title and at least one character that isn't a whitespace. 

Let me know if there is any problem. Thanks for letting me help!

### CODE REVIEW

File: public/js/index.js [hunk 1]
```diff
@@ -294,7 +294,12 @@ async function addLesson() {
 	const isEditView = createLessonContainer
 		.getAttribute('view')
 		.includes('edit-lesson');
-
+	// Regex to match any number of whitespaces in the content form.
+	var regex = /^\<p\>\s+\<\/p\>$/;
+	if (content === "<p><br></p>" || content.match(regex)) {
```
Comment (jrobind) [inline public/js/index.js]: You're on the right track! This pretty much works for all use cases but when I add multiple line breaks it still allows me to create a lesson. There are a couple of approaches I can think of to address this:

1. update the regex to catch multiple line breaks
2. use the `DOMParser` and grab the textContent from the parsed HTML to check if any exists

There are probably more approaches so feel free to explore other options. If you're not sure regarding the `DOMParser` you can check the [docs](https://developer.mozilla.org/en-US/docs/Web/API/DOMParser)

**Review by jrobind — CHANGES_REQUESTED** [review]

**Comment by jrobind** [comment]
Thanks for sending a PR over btw! Don't panic if you mess things up (which you haven't) this is a space for learning and making mistakes 😄 

Let me know if my review doesn't make sense.

Thanks again!

File: public/js/index.js (hunk 1 above)
Comment (edXmO) [reply public/js/index.js]: Ok , so after having a fight with the Regexes,  I noticed that the Quill returns html tags <p></p> for only whitespace cases and <p><br></p> in multiple line breaks. 
So what I did with the new regex is try to get rid of the html tags, then remove the whitespaces and check if the length of the remaining string is still 0, if so, content is empty, no lesson is added. 

I hope I did not mess up anything, still learning to use git and had some trouble with the commits but I think I managed to get the changes through.

File: public/js/index.js [hunk 2]
```diff
@@ -296,6 +296,64 @@ async function addUser(user) {
...
+				});
+
+			lessonInput.value = '';
+			renderLessons();
+		} catch (error) {
+			console.log(error);
+		}
+	}
+
   const content = quill.root.innerHTML;
```
Comment (jrobind) [inline public/js/index.js]: Looks like we have some duplicate code within `addLesson()` that needs removing

File: public/js/index.js [hunk 3]
```diff
@@ -296,6 +296,64 @@ async function addUser(user) {
 }
 
 async function addLesson() {
+	const content = quill.root.innerHTML;
+	const isEditView = createLessonContainer
+		.getAttribute('view')
+		.includes('edit-lesson');
+	// Regex to match any number of whitespaces in the content form.
+	var regex = /<(.|\n)*?>/g;
+	if (content.replace(regex, '').trim().length === 0) {
```
Comment (jrobind) [inline public/js/index.js:305]: Nice job 👍

**Review by jrobind — CHANGES_REQUESTED** [review]
Good job with this! Everything works as expected, except you have some duplicate code within `renderLessons()`. If you can remove that then I'll get this merged in 😄

File: public/js/index.js (hunk 1 above)
Comment (jrobind) [reply public/js/index.js]: Good job figuring out the logic for this. Just left you a small comment on the review. Let me know if it doesn't make sense!

File: public/js/index.js (hunk 2 above)
Comment (edXmO) [reply public/js/index.js]: Done! Removed the duplicate code within addLesson() and commited the changes. 
Let me know if there's anything more.

Comment (jrobind) [reply public/js/index.js]: Perfect!

**Comment by jrobind** [comment]
Thanks @edXmO, merging now 👍

**Pull request merged**